File: LexParser/PalavrasReservadas.cpp
#include "PalavrasReservadas.h"

// Inicializa a tabela de palavra reservadas
PalavrasReservadas::PalavrasReservadas() {
    tabela = {
        {SIMB_PROGRAM,      T_PROGRAM}, 
        {SIMB_BEGIN,        T_BEGIN}, 
        {SIMB_END,          T_END}, 
        {SIMB_VAR,          T_VAR}, 
        {SIMB_CONST,        T_CONST}, 
        {SIMB_REAL,         T_REAL}, 
        {SIMB_INTEGER,      T_INTEGER}, 
        {SIMB_PROCEDURE,    T_PROCEDURE}, 
        {SIMB_READ,         T_READ}, 
        {SIMB_WRITE,        T_WRITE}, 
        {SIMB_IF,           T_IF}, 
        {SIMB_THEN,         T_THEN}, 
        {SIMB_ELSE,         T_ELSE}, 
        {SIMB_FOR,          T_FOR}, 
    };
}

Token PalavrasReservadas::operator[](const string& palavra) {
    Token myToken;
    try {
        type = tabela.at(palavra);
    } catch(out_of_range& ex) { // Não encontrou o id nas palavras reservadas
        type = T_ID;
    }

    return type;
}

File: DFPLalg/src/dfp/PalavrasReservadas.java
package dfp;

import java.util.Hashtable;

public class PalavrasReservadas extends Object {
	 private Hashtable<String, MyToken> tabela;
	 public PalavrasReservadas(){
		tabela = new Hashtable<String, MyToken>();
		tabela.put("program", MyToken.T_PROGRAM);
		tabela.put("begin", MyToken.T_BEGIN);
		tabela.put("end", MyToken.T_END);
		tabela.put("var", MyToken.T_VAR);
		tabela.put("real", MyToken.T_REAL);
		tabela.put("integer", MyToken.T_INTEGER);
		tabela.put("procedure", MyToken.T_PROCEDURE);
		tabela.put("read", MyToken.T_READ);
		tabela.put("write", MyToken.T_WRITE);
		tabela.put("while", MyToken.T_WHILE);
		tabela.put("if", MyToken.T_IF);
		tabela.put("then", MyToken.T_THEN);
		tabela.put("else", MyToken.T_ELSE);
		tabela.put("for", MyToken.T_FOR);
		tabela.put("to", MyToken.T_TO);
		tabela.put("do", MyToken.T_DO);
	 }	 
	 public PalavrasReservadas(Hashtable<String, MyToken> tabela) {
	    this.tabela = tabela;

	 }
	 public Hashtable<String, MyToken> getTabela(){
		  return this.tabela;
	 }
 }
